apps/api/src/certificate/services/certificate.service.ts

```diff
@@ -0,0 +1,49 @@
+import { certificateManager } from "@akashnetwork/akashjs/build/certificates/certificate-manager";
+import { singleton } from "tsyringe";
+
+import { AuthService } from "@src/auth/services/auth.service";
+import { UserWalletRepository } from "@src/billing/repositories";
+import { RpcMessageService } from "@src/billing/services";
+import { ChainErrorService } from "@src/billing/services/chain-error/chain-error.service";
+import { ManagedSignerService } from "@src/billing/services/managed-signer/managed-signer.service";
+
+interface CertificateOutput {
+  certPem: string;
+  pubkeyPem: string;
+  encryptedKey: string;
+}
+
+@singleton()
+export class CertificateService {
+  constructor(
+    private readonly userWalletRepository: UserWalletRepository,
+    private readonly authService: AuthService,
+    private readonly rpcMessageService: RpcMessageService,
+    private readonly managedSignerService: ManagedSignerService,
+    private readonly chainErrorService: ChainErrorService
+  ) {}
+
+  async create(): Promise<CertificateOutput> {
+    const userWallet = await this.userWalletRepository.findOneByUserId(this.authService.currentUser.id);
+
+    if (userWallet) {
+      const { cert: crtpem, publicKey: pubpem, privateKey: encryptedKey } = certificateManager.generatePEM(userWallet.address);
+      const createCertificateMsg = this.rpcMessageService.getCreateCertificateMsg(userWallet.address, crtpem, pubpem);
+      const messages = [createCertificateMsg];
+
+      try {
+        await this.managedSignerService.executeManagedTx(userWallet.id, messages);
```

Comment:
I will change this to `executeDecodedTxByUserId(...)` so it refreshes the user balances as well. (https://github.com/akash-network/console/pull/860/files#r1970537622)